datafusion/common/src/scalar/mod.rs

```diff
@@ -275,6 +277,8 @@ pub enum ScalarValue {
     DurationMicrosecond(Option<i64>),
     /// Duration in nanoseconds
     DurationNanosecond(Option<i64>),
+    /// A nested datatype that can represent slots of differing types. Components:
```

Comment:
The `"Components: " seem` to be cut off in mid-sentence. Perhaps we could document what each of the 4 fields represents?

Reply:
Sorry, this must've gotten wiped in my cherry pick. I restored the rest of the comment.